passkey-client/Cargo.toml

```diff
@@ -23,12 +27,17 @@ public-suffix = { path = "../public-suffix", version = "0.1" }
 serde = { version = "1", features = ["derive"] }
 serde_json = "1"
 ciborium = "0.2"
+mockall = { version = "0.11", optional = true }
 typeshare = "1"
 idna = "0.5"
 url = "2"
 coset = "0.3"
+tokio = { version = "1", features = ["sync"], optional = true }
```

Comment:
**point-of-interest:** I'm not exactly sure why I had to add `mockall` and `tokio` to `passkey-client`, but it was the only way I could get it to compile after I added `#[tokio::test]` in `authenticator.rs`.

Reply:
The reason they need to be added is that they are declared as dependencies for the two added features:
https://github.com/bitwarden/passkey-rs/blob/cea73a27e98ba0acc5b395c29b8d0733a90eec1a/passkey-client/Cargo.toml#L18-L19

Those features seem to me like they aren't used anywhere in the `passkey-client` crate so they can be removed and that means `tokio` and `mockall` can be removed from the `[dependencies]` section.


mockall still needs to be in `[dev-dependencies]` as it's used here:
https://github.com/bitwarden/passkey-rs/blob/cea73a27e98ba0acc5b395c29b8d0733a90eec1a/passkey-client/src/tests/mod.rs#L62-L64

Reply:
Yeah exactly, but if I do that then those lines give me an error, that's what confuses me 

![image](https://github.com/bitwarden/passkey-rs/assets/2285588/a97a96ba-6b95-4704-a076-354aadac6c28)

![image](https://github.com/bitwarden/passkey-rs/assets/2285588/7279d9a5-6133-42d1-bccb-901e63c3c3af)

Reply:
Hmm weird, passkey-authenticator also has them as normal optional dependencies too, so I guess it makes sense to follow that established pattern anyway.